### ISSUE TITLE
Current Version of Crypto is not supported

### ISSUE THREAD

**Issue by ghost** [DrMcCoy/tg3442_firewall#1]
According to [this thread](https://stackoverflow.com/questions/43987779/python-module-crypto-cipher-aes-has-no-attribute-mode-ccm-even-though-pycry), the current version of Crypto doesnt support MODE_CCM anymore and can be replaced with Cryptodome. This works for me.

**Comment by DrMcCoy** [comment]
Oops, I missed this here, sorry about that.

I can't really test this little script anymore, since opted to upgrade to the Fritz!Box, now that it's actually available from Vodafone.

That's pretty annoying, though, that they're removing features like that. :/

Because I can't really test this script anymore, I don't really feel comfortable just replacing the package myself. I can leave this issue open for others to see. Fair?

**Comment by DrMcCoy** [comment]
Oh, and thanks for the heads-up, of course :)

**Comment by DrMcCoy** [comment]
Ah, okay, reading about it more, it's rather that PyCrypto seems to be abandoned. MODE_CCM was introduced with an experimental version in 2013 that barely anybody has picked up.

PyCryptodome seems to be the recommended implementation for it now, acting as a drop-in replacement. It even puts stuff into Crypto.*, like PyCrypto used to do. I too have PyCryptodome installed through the Gentoo package manager, not PyCrypto.

So I guess the correct fix for this issue is not to change anything in script, but people need to just install PyCryptodome instead of PyCrypto.

**Comment by wklaebe** [comment]
On Debian (and Devuan), there's python3-pycryptodome, and Crypto.Cipher exists too (in python3-crypto), but without AES_CCM; so there the change is needed.

**Comment by DrMcCoy** [comment]
Ugh, that's a bit of a mess :/

On Debian, what's the namespace of the crypto with AES_CCM, then? CryptoDome.Cipher?

**Comment by wklaebe** [comment]
Cryptodome.Cipher.
As in:
from Cryptodome.Cipher import AES
print(AES.AES_CCM)

**Comment by DrMcCoy** [comment]
Hmm, here on Gentoo, it's not within Cryptodome, but Crypto.

It seems Cryptodome can be packaged either way, with the version with the Cryptodome namespace being called cryptodomex?

I guess what this script needs, then, is a conditional import. I admit I'm not really that firm in Python (I just use it occasionally, mostly for things like IDAPython or somesuch).

Is there a way to probe for what's available? Or alternatively, would something like

````
try:
    from Cryptodome.Cipher import AES
except ImportError:
    from Crypto.Cipher import AES
````

work? I.e. try to import from Cryptodome, and if that doesn't exist (because Cryptodome has been packaged ito the Crypto namespace instead) use Crypto?

**Comment by wklaebe** [comment]
That's a pattern that I have seen, yes, that should work.

**Comment by DrMcCoy** [comment]
Okay, pushed this as a fix as 27020d8e8ab7ef2053145999b34e18df9d1a4dd6. Thanks! :)

Feel free to poke me if this still doesn't work for you.
